benchmark-ph1ex1/3D-model/postprocessing.py

```diff
@@ -0,0 +1,236 @@
+    print('physical memory usage: {0} MiB'.format(mem))
+
+
+def plot_axial_power_distribution(filename, save):
+    '''
+    This function plots the axial power distribution.
+
+    Parameters:
+    -----------
+    filename: [string]
+        name of the .csv file with results
+    save: [string]
+        name of the figure where to save the plot
+    '''
+
+    file = pd.read_csv(filename)
+
+    power = np.zeros(10)
+    for i in range(10):
+        power[i] = float(file['radially_averaged_axial_power_Z'
+                         + str(i+1)].tolist()[-1])
+
+    z = np.linspace(0, 793, 10, endpoint=False)
+    z += 79.3/2
+
+    plt.figure()
+    plt.plot(z, power)
+    plt.ylim(top=8)
+
+    plt.xticks(fontsize=12)
+    plt.yticks(fontsize=12)
+    plt.ylabel(r'Power density [W/cm$^3$]', fontsize=12)
+    plt.xlabel('Axial distance from bottom of active core [cm]', fontsize=12)
+    plt.grid()
+    plt.savefig(save, dpi=300, bbox_inches="tight")
+
+
+def plot_radial_power_distribution(filename, save):
+    '''
+    This function plots the radial power distribution.
+
+    Parameters:
+    -----------
+    filename: [string]
+        name of the .csv file with results
+    save: [string]
+        name of the figure where to save the plot
+    '''
+
+    file = pd.read_csv(filename)
+
+    power = np.zeros(22)
+    for i in range(22):
+        power[i] = float(file['axially_averaged_radial_power_R'
+                         + str(i+1)].tolist()[-1])
+
+    P = 36  # pitch
+    F = P / np.sqrt(3)
+
+    coord = []
+    # 1 - 6
+    for i in range(4):
+        coord.append(np.array([i*(F+F/2), 3*P-i*P/2]))
+    for i in range(1, 3):
+        coord.append(np.array([3*(F+F/2), 3*P-3*P/2-i*P]))
+
+    # 7 - 14
+    for i in range(5):
+        coord.append(np.array([i*(F+F/2), 4*P-i*P/2]))
+    for i in range(1, 4):
+        coord.append(np.array([4*(F+F/2), 4*P-4*P/2-i*P]))
+
+    # 15 - 22
+    for i in range(1, 5):
+        coord.append(np.array([i*(F+F/2), 5*P-i*P/2]))
+    for i in range(1, 5):
+        coord.append(np.array([5*(F+F/2), 5*P-5*P/2-i*P]))
+
+    coord = np.array(coord)
+
+    plt.figure()
+    patches = []
+    xmax, ymax = [-np.inf, ] * 2
+    xmin, ymin = [np.inf, ] * 2
+    for i in range(len(coord)):
+        h = RegularPolygon(coord[i], 6, F, np.pi/2)
+        patches.append(h)
+        verts = h.get_verts()
+        vmins = verts.min(0)
+        vmaxs = verts.max(0)
+        xmax = max(xmax, vmaxs[0])
+        xmin = min(xmin, vmins[0])
+        ymax = max(ymax, vmaxs[1])
+        ymin = min(ymin, vmins[1])
+
+    patches = np.array(patches, dtype=object)
+    pc = PatchCollection(patches)
+
+    ax = gca()
+    pc.set_array(power)
+    ax.add_collection(pc)
+    ax.set_xlim(xmin, xmax)
+    ax.set_ylim(ymin, ymax)
+
+    cbar = plt.colorbar(pc)
+    cbar.ax.set_ylabel(r'Power density [W/cm$^3$]')
+
+    for i in range(22):
+        plt.text(x=coord[i][0]-F/1.5, y=coord[i][1]-3.5,
```

Comment:
you might benefit from a matplotlibrc file committed to this repository, which will define these parameters for all the plots made in this repo.

Reply:
Yes, this would have been helpful
I will leave it for enhancement